lib/iris/mesh/components.py

```diff
@@ -88,6 +89,19 @@
 )
 
 
+class _Timestamp:
+    """Makes timestamps mutable.
+
+    This behaviour is used to ensure garbage collection works as expected.
```

Comment:
While this is the purpose we're creating a class rather than storing a raw datetime, I think that's the secondary rather than the primary purpose of this class. The primary purpose is to store the last time the associated object has been updated. You could go for something more like

"""Stores the last time the timestamped object has been updated.

The timestamp for a parent object may be held and updated by child objects without the need for the child objects to contain a reference to the parent object. This ought to prevent large parent objects from being held unnecessarily in memory. 
"""